opm/input/eclipse/EclipseState/Grid/Carfin.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+  Copyright 2022 Equinor
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CARFIN_HPP_
+#define CARFIN_HPP_
+
+#include <opm/input/eclipse/EclipseState/Grid/GridDims.hpp>
+
+#include <array>
+#include <cstddef>
+#include <functional>
+#include <vector>
+#include <string>
+
+namespace Opm {
+    class DeckRecord;
+}
+
+namespace Opm
+{
+    class Carfin
+    {
+    public:
+        using IsActive = std::function<bool(const std::size_t globalIdx)>;
+        using ActiveIdx = std::function<std::size_t(const std::size_t globalIdx)>;
+
+        struct cell_index
+        {
+            std::size_t global_index;
+            std::size_t active_index;
+            std::size_t data_index;
+
+            cell_index(std::size_t g,std::size_t a, std::size_t d)
+                : global_index(g)
+                , active_index(a)
+                , data_index(d)
+            {}
+
+            cell_index(std::size_t g, std::size_t d)
+                : global_index(g)
+                , active_index(g)
```

Comment:
I think we should use some marker here, to somehow indicate that this is inactive, e.g.
```
#include<numeric_limits>
....
, active_index(std::numeric_limits<std::size_t>::max())
```

Reply:
> I think we should use some marker here, to somehow indicate that this is inactive, e.g.

I suppose that's debatable.  This particular constructor is used here to initialise the "global" index list and it's not clear to me that marking the cell as inactive is the right thing to do.